app/services/forest_liana/search_query_builder.rb

```diff
@@ -81,15 +84,24 @@ def search_param
 
         if (@params['searchExtended'].to_i == 1)
           SchemaUtils.one_associations(@resource).map(&:name).each do |association|
+            if @collection.search_fields
+              association_search = @collection.search_fields.map do |field|
+                if field.include? association.to_s
```

Comment:
For a more secure code you should check that:
```ruby
if field.include? '.' && field.split('.')[0] == association.to_s
  field.split('.')[1]
end
```

Reply:
I am also pretty sure that `association_search` looks like an array with `nil` values and should not for better performances.
![screen shot 2017-09-06 at 09 22 37](https://user-images.githubusercontent.com/664261/30099142-160dd168-92e5-11e7-9081-cf461565df9b.png)